pvlib/iotools/epw.py

```diff
@@ -225,14 +225,19 @@ def read_epw(filename, coerce_year=None):
             'AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.87 '
             'Safari/537.36')})
         response = urlopen(request)
-        csvdata = io.StringIO(response.read().decode(errors='ignore'))
+        with io.StringIO(response.read().decode(errors='ignore')) as csvdata:
+            try:
+                data, meta = parse_epw(csvdata, coerce_year)
+            except Exception as e:
+                print(e)
     else:
         # Assume it's accessible via the file system
-        csvdata = open(str(filename), 'r')
-    try:
-        data, meta = parse_epw(csvdata, coerce_year)
-    finally:
-        csvdata.close()
+        with open(str(filename), 'r') as csvdata:
+            try:
+                data, meta = parse_epw(csvdata, coerce_year)
+            except Exception as e:
+                print(e)
```

Comment:
I think we definitely want to propagate any parsing errors instead of printing them and letting execution continue, so better to get rid of the `try/except`s in these two blocks.

Reply:
Thanks @chrisorner for the contribution and closing a long-standing annoying issue, and also thanks @kanderso-nrel for quick and thoughtful review. I apologize for commenting after the fact. I agree with Kevin (altho he might not agree with me). In my opinion we should restrict using `print()` in pvlib, except perhaps in examples, b/c it blocks the main execution thread which reduces performance. An alternative is [`logging`](https://docs.python.org/3/library/logging.html) which has many useful features, executes in separate thread, and therefore, does not block the main Python thread. However, as a rule pvlib has avoided logging except during development. Hope this makes sense to folks. Just my thoughts/opinions, definitely do not speak for anyone other than myself.

Reply:
@mikofski just to be sure, would you change anything about the state of this PR as it was merged?  Note that this printing was removed in a later commit and didn't make it into the final squashed diff.

I definitely agree that `print()` inside pvlib itself is never (or at least very rarely) the best way to present information to the user.

Reply:
Nope, no changes. I was very happy with this PR and the review. Thanks again! Just wanted to make sure it was in the record that I extremely discourage `print()`, and I consider `logging` a better alternative although to date, hasn't been used in pvlib much.